src/content/platform-integration/ios/app-extensions.md

```diff
@@ -117,207 +217,361 @@ Choose one of the following sources for your data.
-                withError: NSError(domain: Bundle.main.bundleIdentifier!, code: 0))
-        }
-    }
+    ```console
+    open ios/Runner.xcworkspace
     ```
-:::note
-1. If you _don't_ cancel the request in `viewDidDisappear`, 
-the `ShareViewController` won't be deallocated after being dismissed. 
 
-2. You must pass in the bundle identifier as the error domain, otherwise it crashes with an `EXC_BAD_INSTRUCTION` error. 
-:::
+1.  In Xcode, disable user script sandboxing.
 
-## Advanced scrolling behavior
+    *   Open the **project navigator**
+        (**View** > **Navigators** > **Project**).
 
-By default, flutter view does not handle scroll gestures in a share extension. 
-To support a scrollable list in the share extension, follow [this instruction][]. 
+    *   In the main window under **TARGETS**, select
+        **ShareExtension**.
 
-[this instruction]: {{site.github}}/flutter/website/issues/164670#issuecomment-2740702986
+    *   Open the **Build Settings** tab.
 
-## Test extensions
+    *   Navigate to **Build Options**.
 
-Testing extensions on simulators and physical devices
-have slightly different procedures.
+    *   Set **User Script Sandboxing** as **No**.
 
-{% comment %}
-The different procedures are necessary due to bugs(which bugs?) in Xcode.
-Revisit these docs after future Xcode releases to see if they are fixed.
-{% endcomment -%}
+1.  In Xcode, add the pre-action to the
+    `ShareExtension` scheme.
 
-### Test on a simulator
+    *   Open the **Manage Schemes** window
+        (**Product** > **Scheme** > **Manage Schemes**).
 
-1. Build and run the main application target.
-1. After the app is launched on the simulator,
-   press <kbd>Cmd</kbd> + <kbd>Shift</kbd> + <kbd>H</kbd>
-   to minimize the app, which switches to the home screen.
-1. Launch an app that supports the share extension,
-   such as the Photos app.
-1. Select a photo, tap the share button, then tap
-   on the share extension icon of your app.
+    *   Select the **ShareExtension** scheme and edit it.
 
-### Test on a physical device
+    *   Expand the **Build** tab.
 
-You can use the following procedure or the
-[Testing on simulators](#test-on-a-simulator) instructions
-to test on physical devices.
+    *   Select **Pre-actions**.
 
-1. Launch the share extension target.
-1. In the popup window that says "Choose an app to run",
-   select an app that can be used to test share extension,
-   such as the Photos app.
-1. Select a photo, tap the share button,
-   then tap on the share extension icon of your app.
+    *   Click **+** and select **New Run Script Action**.
+
+    *   In the **Provide build settings from**
+        drop-down list, select **ShareExtension**.
+
+    *   In the **Shell** text field, enter:
+
+        ```console
+        /bin/sh "$FLUTTER_ROOT/packages/flutter_tools/bin/xcode_backend.sh" prepare
+        ```
+
+    *   Click **Close**.
+
+1.  In Xcode, share the build configurations.
```

Comment:
The old docs had a picture here that is helpful. What do you think about adding it back?

Here's an updated image with the correct names:
![Screenshot 2025-05-13 at 10 22 22 AM](https://github.com/user-attachments/assets/80d2c1dd-4fe1-4f70-b01f-b8362533926f)

Reply:
I was on the fence with this, but I'll go ahead and add a screenshot here.